wd_util.c

```diff
@@ -312,12 +312,76 @@ void dump_env_info(struct wd_env_config *config)
 	}
 }
 
+static void *wd_get_config_numa(struct wd_env_config *config, int node)
+{
+	struct wd_env_config_per_numa *config_numa;
+	int i;
+
+	FOREACH_NUMA(i, config, config_numa)
+		if (config_numa->node == node)
+			break;
+
+	if (i == config->numa_num)
+		return NULL;
+
+	return config_numa;
+}
+
+static void wd_free_numa(struct wd_env_config *config)
+{
+	struct wd_env_config_per_numa *config_numa;
+	int i;
+
+	if (!config->config_per_numa)
+		return;
+
+	FOREACH_NUMA(i, config, config_numa)
+		free(config_numa->dev);
+
+	free(config->config_per_numa);
+}
+
+static void wd_get_dev_numa(struct uacce_dev_list *head,
+			    int *numa_dev_num, int *numa_num)
+{
+	struct uacce_dev_list *list = head;
+
+	while (list) {
+		if (list->dev->numa_id < 0)
+			list->dev->numa_id = 0;
+
+		if (!numa_dev_num[list->dev->numa_id])
+			(*numa_num)++;
```

Comment:
You increased **\*numa_num** directly. But what's the initial/expected value of **\*numa_num**?

Reply:
You can see numa_num initial value is 0 in wd_alloc_numa statement. Add it one time only when we find a device on a new numa node.if device numa node is 0, 0, 2，we add it one only in first and third device

Reply:
It seems that wd_get_dev_numa() is only called once. And **\*numa_num** is initialized out of the wd_get_dev_numa(). Maybe you can initialize it just in wd_get_dev_numa().

Reply:
Resubmit these changes now.